settings.py

```diff
@@ -162,8 +163,13 @@ def update(self, dct: Dict[str, Any]) -> None:
             else:
                 # assign value, try to upcast to type hint
                 annotation = self.get_type_hints().get(k, None)
-                candidates = [] if annotation is None else \
-                    typing.get_args(annotation) if typing.get_origin(annotation) is Union else [annotation]
+                candidates = (
+                    [] if annotation is None else (
+                        typing.get_args(annotation)
+                        if typing.get_origin(annotation) in {Union, types.UnionType}
```

Comment:
```suggestion
                        if typing.get_origin(annotation) in (Union, types.UnionType)
```
For 2 items, `in` tuple is as fast or faster than `in` set, uses less memory and loads/compiles faster.
( https://discord.com/channels/731205301247803413/731214280439103580/1275375346807476235 )

Reply:
I've tested this before, and I was surprised to find that set was faster than tuple for 2 items.
But that was a long time ago, and it's something that could change with different Python versions and platforms.
I didn't test it for this or any time recently.